exporter/collector/metricsexporter.go

```diff
@@ -162,6 +174,121 @@ func (m *metricMapper) metricToTimeSeries(
 	return timeSeries
 }
 
+func (m *metricMapper) exemplar(ex pdata.Exemplar) *distribution.Distribution_Exemplar {
+	attachments := []*anypb.Any{}
+	// TODO: Look into still sending exemplars with no span.
+	if !ex.TraceID().IsEmpty() && !ex.SpanID().IsEmpty() {
+		sctx, err := anypb.New(&monitoringpb.SpanContext{
+			// TODO - make sure project id is correct.
+			SpanName: fmt.Sprintf("projects/%s/traces/%s/spans/%s", m.cfg.ProjectID, ex.TraceID().HexString(), ex.SpanID().HexString()),
+		})
+		if err == nil {
+			attachments = append(attachments, sctx)
+		} else {
+			log.Printf("Failure to write span context: %v", err)
```

Comment:
If we are going to log within the exporter, we should use the `params.Logger` provided to us when the exporter as part of `newGoogleCloudMetricsExporter`.  But I think we should consider dropping the metric if we can't write the exemplar so the failures can be detected using the self-obs metrics (dropped/sent/failed send).

Same comment for error handling below.

Reply:
Two questions here that make me disagree with dropping the point:

1. If my exemplars are broken do I also want my histograms to stop showing up?
2. What are the likely failures where this `err` would be returned? (i.e. failure to locally serialize a proto into an Any).

I agree on logger.

Reply:
I'm not entirely sure about 2.  My hope would be "only if something went terribly wrong", but i'm just guessing.  1 is a good question, but I generally lean towards failing loudly.  Would this be a debug-level log, since it is per-request?  Or would we make it a warning?  If the former, i'm not sure it is enough of a signal for someone to notice if exemplars are broken

Reply:
For 2 - having done SOME inspection, it seem like there are two primary ways this fails:

- If you pass a `nil` message
- If you have a required filed that's missing a value.

I think both are very rare in practice.

For 1 - I'm fine if we fail loudly where it's a logic error.  If it's the case of "we don't know the shape of user data", I think here it's better to give the user as MUCH of a working solution as we can (while loudly complaining).  Failing to write the point is different than loudly complaining.  I think we can accomplish both of our goals here with a different technique (you're right about debug logs).  Let's noodle on this for next time where it's more likely an error could happen at runtime post-unit-tests.

Reply:
Regarding using params.Logger, going to open a new bug about that one, need some wiring/funnelling.

Reply:
sgtm